saltgui/static/scripts/dropdown.js

```diff
@@ -53,13 +56,13 @@ class DropDownMenu {
   // function is called each time the menu opens
   // This allows dynamic menuitem titles (use menuitem.innerText/innerHTML)
   // or visibility (use menuitem.style.display = "none"/"inline-block")
-  addMenuItem(title, callback) {
+  addMenuItem(title, callback, value) {
     const button = Route._createDiv("run-command-button", "...");
     if(typeof title === "string")
       button.innerHTML = title;
     else
       button.verifyCallback = title;
-    button.addEventListener('click', evt => callback(evt));
+    button.addEventListener('click', evt => this.callback(evt, callback, value));
     this.menuDropdownContent.appendChild(button);
```

Comment:
This is a value-selection menu. The selected value is stored in the menu object instance because it cannot be applied immediately. The value is used later, In this case when you click the "Run command" button.
The property is not added to an HTML object, only to the object encapsulating one.